java-dynamic-sqs-listener-core/src/main/java/com/jashmore/sqs/broker/AbstractMessageBroker.java

```diff
@@ -0,0 +1,81 @@
+package com.jashmore.sqs.broker;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+
+/**
+ * Abstract {@link MessageBroker} that handles the starting and stopping of the background broker thread.
+ */
+@Slf4j
+@RequiredArgsConstructor
+public abstract class AbstractMessageBroker implements MessageBroker {
+    private final ExecutorService executorService;
```

Comment:
Having the specific implementations pass in an ExecutorService so that we can run 1 and only 1 task on it - the main controller thready should be optional. I like what you did w/ the single one where a simple single (2) thread pool was passed in.

Reply:
Ah so if they don't want to pass in their own ExecutorService it should optionally just use a cachedThreadPool or something? Is that what you are saying?